### PR TITLE
[DOC] Clarify relationshi load_best_model_at_end and save_total_limit

### PR DESCRIPTION
**Pull request by BramVanroy**
Clarify the relationship between `load_best_model_at_end` and `save_total_limit`. Hope this is clear.

As discussed on Slack @sgugger

### CODE REVIEW

File: src/transformers/training_args.py [hunk 1]
```diff
@@ -283,7 +283,11 @@ class TrainingArguments:
             float in range `[0,1)`. If smaller than 1, will be interpreted as ratio of total training steps.
         save_total_limit (`int`, *optional*):
             If a value is passed, will limit the total amount of checkpoints. Deletes the older checkpoints in
-            `output_dir`.
+            `output_dir`. When `load_best_model_at_end` is enabled, the "best" checkpoint according to
+            `metric_for_best_model` will always be retained in addition to the most recent ones. For example, for
+            `save_total_limit=5` and `load_best_model_at_end`, the four last checkpoints will always be retained
+            alongside the best model. When `save_total_limit=1` and `load_best_model_at_end`, it is possible that two
+            checkpoints are saved: the last one and the best one (if they are different).
```
Comment (sgugger) [inline src/transformers/training_args.py:290]: I would put `load_best_model_at_end=True` in the last two instances.

File: src/transformers/training_args.py [hunk 2]
```diff
@@ -371,7 +375,8 @@ class TrainingArguments:
             except if the model used is one of the `XxxForQuestionAnswering` in which case it will also include the
             `["start_positions", "end_positions"]` keys.
         load_best_model_at_end (`bool`, *optional*, defaults to `False`):
-            Whether or not to load the best model found during training at the end of training.
+            Whether or not to load the best model found during training at the end of training. When this option is
+            enabled, the best checkpoint will always be saved. See `save_total_limit` for more.
```
Comment (sgugger) [inline src/transformers/training_args.py:379]: Nit, you can link to the doc of `save_total_limit`:
```suggestion
            enabled, the best checkpoint will always be saved. See [`save_total_limit`](https://huggingface.co/docs/transformers/main_classes/trainer#transformers.TrainingArguments.save_total_limit) for more.
```

**Review by sgugger — COMMENTED** [review]
Thanks a lot for documenting this better! I left a couple of comments.

File: src/transformers/training_args.py (hunk 1 above)
Comment (BramVanroy) [reply src/transformers/training_args.py:290]: Done!

**Comment by sgugger** [comment]
Can you just do a quick rebase/merge on the main branch? I think the issue in the tests (due to a release of Pillow apparently) is fixed on main.

**Review by sgugger — APPROVED** [review]

**Pull request merged**